addons/website_sale/models/product.py

```diff
@@ -91,6 +91,17 @@ def _get_website_pricelists_domain(self, website_id):
             '|', ('selectable', '=', True), ('code', '!=', False),
         ]
 
+    def _get_partner_pricelist_multi(self, partner_ids, company_id=None):
+        ''' If `property_product_pricelist` is read from website, we should use
+            the website's company and not the user's one.
+            Passing a `company_id` to super will avoid using the current user's
+            company.
+        '''
+        website = ir_http.get_request_website()
+        if not company_id and website:
+            company_id = website.company_id.id
```

Comment:
@nim-odoo From my tests no error on python reload with this PR without an update.
Except this case:
What might happen is that user won't get the new `ir.rule` on company pricelists to be able to read pricelists from website's company in addition to his company ones.

In that case, a specific setup might lead to an access error: A partner on company 1 which has a pricelist from company 1 specifically set on him. Then if that partner visit a website which is from another company (not allowed for that partner) it will raise an error as no access to company 2 without the new ir.rules.

We could add a condition here, a try/except to be sure the user can access the pricelists from the website's company.
If the try fails, it means the module was not updated and we can't use the website's company.

Something like
```python
if not company_id and website:
    # Compatibility check, `company_id = website.company_id.id` has to be
    # conditionned. Indeed new ir.rule to access pricelists from website's
    # company might not exists (reload python without module update).
    # Remove me in saas-12.1
    try:
        website.company_id.check_access_rule('read')
        company_id = website.company_id.id
    except:
        pass
```

Note that currently you already have access error on standard code when setuping multi-company/multi-website environment, but mostly 403 and not 500.

Reply:
Hum, I don't get it. The new rules apply to product.pricelist and product.pricelist.item, so why would it fail on website.company_id.id? Anyway that's better to always foresee a fallback in these cases. It's also better to test except AccessError

Reply:
Actually reading the `id` `website.company_id.id` will not raise any traceback even if the user can't access the company. I could have tried to read `website.company_id.name` instead of `website.company_id.check_access_rule('read')`.

But if we pass `website.company_id.id` as company and the user does not have the new `ir.rule`, it will throw an error since:
1. We will try to read `property_product_pricelist` which is a compute
2. that is calling `_get_partner_pricelist_multi` which does
```python
Property = self.env['ir.property'].with_context(force_company=company_id or self.env.user.company_id.id)
result = Property.get_multi('property_product_pricelist', Partner._name, partner_ids)
```

Then it would return the pricelists from the website company, and if the new `ir.rule` are not there, it will raise an error if the website's company is not the user's company.

Reply:
Indeed. I correctly fixed the compatibility issue.
The case is still:

> A partner on company 1 which has a pricelist from company 1 specifically set on him. Then if that partner visit a website which is from another company (not allowed for that partner) it will raise an error as no access to company 2 pricelists without the new ir.rules.

I pushed a commit that I will amend if it looks ok to you. You can directly check it.
Basically, I try to read the website's company's pricelists, if it raise an error, it means the DB did not update `website_sale` and does not have the new `ir.rule` allowing website's company pricelists to be accessed. In that case, I just fallback on old implementation.

Reply:
ok :smile: